### ISSUE TITLE
Pin sentry-android Dependency

### ISSUE THREAD

**Issue by davidofwatkins** [getsentry/sentry-cordova#158]
At some point recently, a new release of the `sentry-android` dependency used in this plugin was published, and suddenly my Cordova app would no longer build without any changes from me. While there is a separate issue having to do with the latest `sentry-android` release or a conflict in my own dependencies, the reason this break occurred so out-of-nowhere for me is because `sentry-cordova` always takes the latest version of `sentry-android`:

https://github.com/getsentry/sentry-cordova/blob/03404f163714271a813610eb53b2318d43cc4d23/plugin.xml#L37

This seems dangerous to me. I think the `sentry-android` version should be pinned here and intentionally updated when a new `sentry-cordova` is released, as such:

```diff
- <framework src="io.sentry:sentry-android:1+" />
+ <framework src="io.sentry:sentry-android:1.26.+" />
```

### For context about my specific problem...

I think this dependency should be better pinned regardless of the error I'm experiencing with `io.sentry:sentry-android:1+`. However, in case it helps anyone else out, I thought I'd include it here for some context:

```
Exception in thread "main" java.lang.IllegalArgumentException
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at com.google.devtools.build.android.desugar.CoreLibraryRewriter.reader(CoreLibraryRewriter.java:44)
	at com.google.devtools.build.android.desugar.Desugar.desugarClassesInInput(Desugar.java:388)
	at com.google.devtools.build.android.desugar.Desugar.desugarOneInput(Desugar.java:326)
	at com.google.devtools.build.android.desugar.Desugar.desugar(Desugar.java:280)
	at com.google.devtools.build.android.desugar.Desugar.main(Desugar.java:584)


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithDesugarForRelease'.
> com.android.build.api.transform.TransformException: java.lang.RuntimeException: java.lang.RuntimeException: com.android.ide.common.process.ProcessException: Error while executing java process with main class com.google.devtools.build.android.desugar.Desugar with arguments {--input /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/74.jar --output /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/desugar/release/76.jar --input /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/69.jar --output /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/desugar/release/71.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/CordovaLib/build/intermediates/intermediate-jars/release/classes.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/classes/release --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/0.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/1.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/2.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/3.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/4.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/5.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/6.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/7.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/8.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/9.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/10.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/11.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/12.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/13.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/14.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/15.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/16.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/17.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/18.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/19.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/20.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/21.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/22.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/23.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/24.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/25.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/26.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/27.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/28.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/29.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/30.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/31.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/32.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/33.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/34.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/35.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/36.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/37.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/38.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/39.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/40.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/41.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/42.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/43.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/44.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/45.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/46.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/47.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/48.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/49.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/50.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/51.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/52.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/53.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/54.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/55.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/56.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/57.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/58.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/59.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/60.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/61.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/62.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/63.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/64.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/65.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/66.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/67.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/68.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/69.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/70.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/71.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/72.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/73.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/74.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/75.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/76.jar --classpath_entry /home/circleci/project/zenput/static/phonegap/platforms/android/app/build/intermediates/transforms/stackFramesFixer/release/77.jar --bootclasspath_entry /opt/android/sdk/platforms/android-28/android.jar --bootclasspath_entry /opt/android/sdk/platforms/android-28/optional/org.apache.http.legacy.jar --bootclasspath_entry /opt/android/sdk/platforms/android-28/optional/android.test.mock.jar --bootclasspath_entry /opt/android/sdk/platforms/android-28/optional/android.test.base.jar --bootclasspath_entry /opt/android/sdk/platforms/android-28/optional/android.test.runner.jar --bootclasspath_entry /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/resources.jar --bootclasspath_entry /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/rt.jar --bootclasspath_entry /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jsse.jar --bootclasspath_entry /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jce.jar --bootclasspath_entry /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/charsets.jar --min_sdk_version 19 --nodesugar_try_with_resources_if_needed --desugar_try_with_resources_omit_runtime_classes}
```

Here's my temporary fix in my own `build-extras.gradle` file until I can address the error and/or `sentry-cordova` pins sentry-android more specifically:

```gradle
configurations.all {
    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
        def requested = details.requested

        if (requested.group == 'io.sentry' && requested.name.startsWith('sentry-android')) {
            details.useVersion '1.7.26'
        }
    }
}
```

**Comment by Kiludude** [comment]
I've got the exact same issue. Where did you put your build-extras.gradle file? I've tried putting it in platforms/android but still got the same error.

**Comment by davidofwatkins** [comment]
@Kiludude I have mine in `platforms/android/app/`.

**Comment by Kiludude** [comment]
Thanks for replying. Doesn't seem to work for me though, get the exact same error. Guess I'll have to use another implementation.

**Comment by davidofwatkins** [comment]
Closing this as it appears to be addressed by #164 🎉
